Source/RapyutaSimulationPlugins/Private/Tools/SimulationState.cpp

```diff
@@ -240,32 +242,70 @@ void ASimulationState::AttachSrv(UROS2GenericSrv* Service)
     AttachService->SetResponse(Response);
 }
 
+AActor* ASimulationState::SpawnEntity(const FROSSpawnEntityRequest& InROSSpawnRequest,
+                                      const TSubclassOf<AActor>& InEntityClass,
+                                      const FTransform& InEntityTransform)
+{
+    // SpawnActorDeferred to set parameters beforehand
+    AActor* newEntity = GetWorld()->SpawnActorDeferred<AActor>(InEntityClass, InEntityTransform);
+    if (nullptr == newEntity)
+    {
+        return nullptr;
+    }
+
+    // Wrap [InROSSpawnRequest] into a ROS2 spawnable component
+    UROS2Spawnable* spawnableComponent = NewObject<UROS2Spawnable>(newEntity, TEXT("ROS2 Spawn Parameters"));
+    spawnableComponent->RegisterComponent();
+    spawnableComponent->InitializeParameters(InROSSpawnRequest);
+    newEntity->AddInstanceComponent(spawnableComponent);
+    newEntity->Rename(*InROSSpawnRequest.StateName);
+#if WITH_EDITOR
+    newEntity->SetActorLabel(*InROSSpawnRequest.StateName);
+#endif
+
+    // Add tags
+    for (auto& tag : InROSSpawnRequest.Tags)
+    {
+        newEntity->Tags.Emplace(tag);
+    }
+
+    // Finish spawning Entity
+    UGameplayStatics::FinishSpawningActor(newEntity, InEntityTransform);
+
+    // Add to [Entities]
+    AddEntity(newEntity);
+    return newEntity;
+}
+
 void ASimulationState::SpawnEntitySrv(UROS2GenericSrv* Service)
 {
     UROS2SpawnEntitySrv* SpawnEntityService = Cast<UROS2SpawnEntitySrv>(Service);
 
-    FROSSpawnEntityRequest Request;
-    SpawnEntityService->GetRequest(Request);

```

Comment:
I typically avoid ambiguous var name regardless of the context or its class type (like if you have more than 2 ones, the sources would be come a bit cryptic). But in this case, since it's not a long function so you are right👍🏼